### PR TITLE
Add New Bitwise Int Operators

### LINKED ISSUE #346: New operators: bitwise int operations
New `EventSet.and()`, `EventSet.or()`, `EventSet.xor()`, `EventSet.not()`, `EventSet.left_shift()`, `EventSet.right_shift()` operators that implement bitwise operations on integer types, with their corresponding magics: `&`, `|`, `^`, `~`, `<<`, `>>`.

`and()` and `or()` are already implemented for bool and need to be extended to work with ints.

See https://github.com/google/temporian/blob/main/CONTRIBUTING.md#developing-a-new-operator for guidance. 

Questions or requests for additional guidance from possible contributors more than welcome!

### PR DESCRIPTION
**Pull request by saneens**
Closes #346. Bitwise OR, Bitwise AND, Bitwise XOR, Left Shift, Right Shift, and Bitwise Invert operators added.

### CODE REVIEW

File: temporian/core/event_set_ops.py [hunk 1]
```diff
@@ -1613,6 +1742,222 @@ def begin(self: EventSetOrNode) -> EventSetOrNode:
...
+
+        self._raise_error("bitwise_and", other, "int")
+        assert False
+
+    def bitwise_or(self: EventSetOrNode, other: Any) -> EventSetOrNode:
+        """Computes the bitwise OR between an [`EventSet`][temporian.EventSet]
+        or a scalar value to `self` element-wise. The operation is only
+        supported for Events with features that are integers.
+
+        If an EventSet, each feature in `self` is bitwise ORed to the feature
```
Comment (javiber) [inline temporian/core/event_set_ops.py:1822]: I'm not convinced about using "bitwise or" as a verb here, I would rephrase around

```suggestion
        If an EventSet, the bitwise or is calculated between each feature in `self` and the feature
```
If we do, `and` and `xor` need to be changed as well

File: temporian/core/operators/test/test_bitwise.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    def test_correct_right_shift(self) -> None:
+        """Test correct RIGHT SHIFT operator."""
+        expected = event_set(
+            timestamps=[1, 2, 3, 4],
+            features={"right_shift_x_x": [21, 21, 7, 0]},
+            same_sampling_as=self.evset_1,
+        )
+        assertOperatorResult(
+            self, right_shift(self.evset_1, self.evset_2), expected
+        )
```
Comment (javiber) [inline temporian/core/operators/test/test_bitwise.py:95]: add one test with incorrect dtypes expecting an exception

Comment (javiber) [reply temporian/core/operators/test/test_bitwise.py:95]: add one test with int32 and int64, the return type should be int64
make sure to include numbers bigger than int32 as done here: https://github.com/google/temporian/blob/main/temporian/core/operators/test/test_cast.py#L24

File: temporian/core/operators/test/test_bitwise_scalar.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+from temporian.core.operators.scalar import (
+    bitwise_and_scalar,
+    bitwise_or_scalar,
+    bitwise_xor_scalar,
+    left_shift_scalar,
+    right_shift_scalar,
+)
+
+
+class ArithmeticScalarTest(absltest.TestCase):
```
Comment (javiber) [inline temporian/core/operators/test/test_bitwise_scalar.py:28]: update class name and docstring

**Review by javiber — CHANGES_REQUESTED** [review]
This is great! I left you some small requests

File: temporian/core/event_set_ops.py [hunk 4]
```diff
@@ -556,34 +556,24 @@ def __neg__(self: EventSetOrNode) -> EventSetOrNode:
         return multiply_scalar(input=self, value=-1)
 
     def __invert__(self: EventSetOrNode) -> EventSetOrNode:
-        """Inverts a boolean [`EventSet`][temporian.EventSet] element-wise.
+        """Inverts a boolean or integer [`EventSet`][temporian.EventSet]
```
[the remark this answers is not in the document]
Comment (saneens) [reply temporian/core/event_set_ops.py:559]: Should bitwise_invert support boolean or just integer types?
